### PR TITLE
fix: permit grifting issue while using permit functionality

### PR DESCRIPTION
**Pull request by parth-15**

### CODE REVIEW

File: src/BaseTokenWrapper.sol [hunk 1]
```diff
@@ -57,15 +57,21 @@ abstract contract BaseTokenWrapper is Ownable, IBaseTokenWrapper {
...
+        msg.sender,
+        address(this),
+        amount,
+        signature.deadline,
+        signature.v,
+        signature.r,
+        signature.s
+      )
+    {
+      //explicitly left blank to protect users from permit grifting
```
Comment (miguelmtzinf) [inline src/BaseTokenWrapper.sol:71]: you can move this comment just before the try.catch, and have the rest of the try-catch in 1 line.
seems cleaner to me

File: src/BaseTokenWrapper.sol [hunk 2]
```diff
@@ -57,15 +57,18 @@ abstract contract BaseTokenWrapper is Ownable, IBaseTokenWrapper {
...
-    IERC20WithPermit(TOKEN_IN).permit(
-      msg.sender,
-      address(this),
-      amount,
-      signature.deadline,
-      signature.v,
-      signature.r,
-      signature.s
-    );
+    //explicitly left try-catch block blank to protect users from permit grifting
```
Comment (miguelmtzinf) [inline src/BaseTokenWrapper.sol:60]:
```suggestion
    // explicitly left try-catch block blank to protect users from permit griefing
```

File: src/BaseTokenWrapper.sol [hunk 3]
```diff
@@ -85,15 +88,18 @@ abstract contract BaseTokenWrapper is Ownable, IBaseTokenWrapper {
...
-    aTokenOut.permit(
-      msg.sender,
-      address(this),
-      amount,
-      signature.deadline,
-      signature.v,
-      signature.r,
-      signature.s
-    );
+    //explicitly left try-catch block blank to protect users from permit grifting
```
Comment (miguelmtzinf) [inline src/BaseTokenWrapper.sol:91]:
```suggestion
    // explicitly left try-catch block blank to protect users from permit griefing
```

File: test/BaseTokenWrapper.t.sol [hunk 4]
```diff
@@ -425,6 +525,80 @@ abstract contract BaseTokenWrapperTest is Test {
     );
   }
 
+  function testPermitGriftingWithdrawTokenWithPermit() public {
```
Comment (miguelmtzinf) [inline test/BaseTokenWrapper.t.sol:528]:
```suggestion
  function testPermitGriefingWithdrawTokenWithPermit() public {
```

**Review by miguelmtzinf — APPROVED** [review]

**Review by sakulstra — APPROVED** [review]

**Comment by miguelmtzinf** [comment]
Thanks @sakulstra for your review.

**Pull request merged**